File: greedy/pro_1.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 0;
    sort(routes.begin(), routes.end());
    int start=routes[0][0]; int end=routes[0][1];
    for(int i=1; i<routes.size(); i++){
        if(routes[i][0]<=end){
            start=routes[i][0];
        }
        if(routes[i][1]<=end){
            end=routes[i][1];
        }
        if(end<routes[i][0]){
            answer++;
            start=routes[i][0];
            end=routes[i][1];
        }
    }
    return answer+1;
}

File: 15th/pro_1.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int zero=0;
        int sucess=0;
        Arrays.sort(win_nums);
        for(int n : lottos){
            if(n==0){
                zero++;
                continue;
            }
            if(Arrays.binarySearch(win_nums,n)>=0)
                sucess++;
        }
        answer[1]=6-sucess+1;
        answer[0]=6-sucess-zero+1;
        for(int i=0; i<2; i++){
            if(answer[i]==7)    answer[i]=6;
        }
        return answer;
    }
}